RandBLAS/random_gen.hh

```diff
@@ -15,22 +17,103 @@
 #include <Random123/boxmuller.hpp>
 #include <Random123/uniform.hpp>
 
-/// @cond
-/// extend r123::float2 to work with structured bindings
-namespace std {
-  template<> struct tuple_size<r123::float2> { static constexpr size_t value = 2; };
-  template<> struct tuple_element<0, r123::float2> { using type = float; };
-  template<> struct tuple_element<1, r123::float2> { using type = float; };
+/// our extensions to random123
+namespace r123ext
```

Comment:
I _love_ this convention for organizing mild add-ons to Random123 functionality.